pkg/service/controller/servingcert/secret_updating_controller.go

```diff
@@ -231,10 +232,23 @@ func (sc *ServiceServingCertUpdateController) syncSecret(key string) error {
 	}
 	secret := t.(*kapi.Secret)
 
-	dnsName := secret.Annotations[ServiceNameAnnotation] + "." + secret.Namespace + ".svc"
+	if len(secret.Annotations[ServiceNameAnnotation]) == 0 {
+		return nil
+	}
+	serviceObj, exists, err := sc.serviceCache.GetByKey(secret.Namespace + "/" + secret.Annotations[ServiceNameAnnotation])
+	if err != nil || !exists {
+		return nil
```

Comment:
Crap... I just realized all this logic was already in regenerate.  Simplifying this again.